app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CategoriesController < ApplicationController
+
+  def index
+    @categories = Category.all
+  end
+
+  def show
+    @category = Category.find(params[:id])
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      flash[:success] = "#{@category.name} category added!"
+      redirect_to category_path(@category)
+    else
+      flash[:fail] = "You must enter a unique Category!"
+      render :new
+    end
+  end
+
+  def edit
+    @category = Category.find(params[:id])
+  end
+
+  def update
+    @category = Category.find(params[:id])
+    @category.update(category_params)
```

Comment:
what happens if a user updates a category to an existing category? will the change save? how can we show our user nice error messages?